src/test/java/cat/udl/eps/softarch/demo/steps/CreateTransDefs.java

```diff
@@ -0,0 +1,77 @@
+package cat.udl.eps.softarch.demo.steps;
+
+import cat.udl.eps.softarch.demo.domain.Transaction;
+import cat.udl.eps.softarch.demo.repository.TransactionRepository;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.junit.Assert;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+
+import java.math.BigDecimal;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+public class CreateTransDefs {
+    @Autowired
+    private TransactionRepository transactionRepository;
+    @Autowired
+    private StepDefs stepDefs;
+
+
+    @And("There is no transaction created")
+    public void thereIsNoTransactionCreated() {
+        Assert.assertEquals(0, transactionRepository.count());
+    }
+
+    @When("I Create a new Transaction")
+    public void iCreateANewTransaction() throws Exception {
+        Transaction transaction = new Transaction();
+
+        stepDefs.result = stepDefs.mockMvc.perform(
+                        post("/transactions")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(stepDefs.mapper.writeValueAsString(transaction))
+                                .accept(MediaType.APPLICATION_JSON)
+                                .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+    @Then("I put the Transaction price {string}")
+    public void iModifyTheTransactionPrice(String price) throws Exception {
+        Transaction transaction = transactionRepository.findById(1L).get();
```

Comment:
It's better not to fix a particular identifier, as these might change depending on the database or test order. The recommended approach is to create the transaction with a specific price just with one step and store the identifier of the new instance using a global variable, like in: 

https://github.com/UdL-EPS-SoftArch/UniCloud-API/blob/main/src/test/java/cat/udl/eps/softarch/unicloud/steps/CreateSubjectStepDefs.java#L53

You can then use that identifier when checking in the @Then if the transaction has been created with the proper price, also like in the example